cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -1145,8 +1153,13 @@ std::unique_ptr<column> merge_tdigests(tdigest_column_view const& tdv,
   auto merged =
     cudf::detail::concatenate(tdigest_views, stream, rmm::mr::get_current_device_resource());
 
+  auto merged_weights = merged->get_column(1).view();
+  // If there are no values, we can simply return a column that has only empty tdigests.
+  if (merged_weights.size() == 0) {
+    return cudf::tdigest::detail::make_tdigest_column_of_empty_clusters(num_groups, stream, mr);
+  }
```

Comment:
Should be just `auto const merged_weights_size = merged->get_column(1).size();`. No need to get a view.

Reply:
Not even needed to create a temp variable and it can directly be used in `if`

Reply:
The `merged_weights` variable is an existing variable and used in other places as well. I just have moved it from its original line. That being said, I can remove it if it's a problem. I thought creating a column view is cheap. Am I missing something?

Reply:
A `column_view`  is not a plain type. It has several internal structures such as vector, pointer, counter etc. Creating an instance will require copying/initializing all these fields. That can be "cheap" but still incur some overhead thus we should avoid doing so if possible.

Reply:
Thanks @ttnghia. Your explanation aligns with my understanding. In that case, I think this is OK because the shortcut in this `if` clause will be used only when every key has the empty cluster. This is quite a rare case and won't happen often in production.